assets/shaders/viewport_render.wgsl

```diff
@@ -617,13 +584,22 @@ fn get_by_ray(ray_: Line) -> OctreeRayIntersection{
                     current_bounds,
                     node_stack[node_stack_i - 1].target_octant
                 );
-                target_child_key = current_node.children[target_octant];
-
+                target_child_key = children_buffer[current_node.children_starts_at + target_octant];
                 if (
                     (!cube_contains_point(current_bounds, node_stack[node_stack_i - 1].child_center))
                     || (
-                      target_child_key < node_count //crate::object_pool::key_is_valid
-                      && is_bitmap_occupied_at_octant(current_node.sized_node_meta, target_octant)
+                        target_child_key < node_count //crate::object_pool::key_is_valid
+                        && 0 != (
+                            get_node_occupancy_bitmap( current_node.sized_node_meta )
+                            & (0x00000001u << (target_octant & 0x000000FF)) // crate::spatial::math::octant_bitmask
+                        )
+                        && 0 != (
+                            get_node_occupancy_bitmap(nodes[target_child_key].sized_node_meta)
+                            & RAY_TO_NODE_OCCUPANCY_BITMASK_LUT[hash_region(
+                                point_in_ray_at_distance(ray, ray_current_distance) - target_bounds.min_position,
+                                target_bounds.size
+                            )][direction_lut_index]
+                        )
```

Comment:
By checking if the target child would also collide with the ray based on its occupancy bitmap, some context switches can be spared. The calculation however is UGLY, but I found no other way to only calculate this when needed
I assume if the previous condition is false, then this statement won't get evaluated, thus the calculations in it will not take up unnecessary compute.